project-koku/koku
# Cost 689 presto image

## Jira Ticket

[COST-689](https://issues.redhat.com/browse/COST-689)

## Description

This ticket makes some changes to the localdev deploy of presto to use the cost-management-managed images `ubi-trino` and `ubi-hive`.

There are some minor changes to the code to help make the (eventual) transition to Trino easier in terms of a migration from PRESTO* env vars to TRINO equivalents.

## Testing

Docker-compose
* Setup environment to process parquet files with presto:
```
export ENABLE_PARQUET_PROCESSING=True
export S3_BUCKET_NAME=koku-bucket
export S3_ENDPOINT=http://kokuminio:9000
export S3_ACCESS_KEY=kokuminioaccess
export S3_SECRET=kokuminiosecret
```
* If presto is running, shut it down: `make docker-presto-down`
* If koku is running, shut it down: `make docker-down && make remove-db && make docker volume prune -f`
* Restart koku and create test customer: `make docker-up && sleep 5 && make docker-reinitdb-with-sources`
* Wait about 20 seconds for all services to be ready.
* Start presto containers: `make docker-presto-up`
* Load some test data: `make load-test-customer-data`
* Get some coffee.
* Eventually, you should be able to logon to presto `presto --server=localhost:8080 --catalog hive --schema acct10001 --user=admin`
* Execute `show tables;` and you should see some tables. If you select from some tables, you should see data.
* If you login to the PostgreSQL database, you should see the corresponding daily summary tables also have data.

## Companion PR

https://github.com/RedHatInsights/e2e-deploy/pull/2622
